### PR TITLE
Complete User Stories 13-16: login/logout functionality

### LINKED ISSUE #13: User Story 13, User can Login
As a visitor
When I visit the login path
I see a field to enter my email address and password
When I submit valid information
If I am a regular user, I am redirected to my profile page
If I am a merchant user, I am redirected to my merchant dashboard page
If I am an admin user, I am redirected to my admin dashboard page
And I see a flash message that I am logged in

### PR DESCRIPTION
**Pull request by DanielEFrampton**
## Description
Resolves #13, #14, #16, #18. (<- numbers don't match user stories for some reason)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## RSpec Results
```
Finished in 3.22 seconds (files took 2.31 seconds to load)
115 examples, 0 failures, 1 pending

Coverage report generated for RSpec to /Users/dframpton/turing/2module/group_projects/monster_shop_part_1/coverage. 1
434 / 1436 LOC (99.86%) covered.
```
(less than 100% because one test is skipped which was pulled from half-finished User Story 3; should pass after merging with that completed branch.)

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    unless login_attempt(params)
+      flash[:bad_credentials] = 'Avast! Thar be problems with the credentials ye input!'
+      redirect_to '/login'
+    end
+  end
+
+  def destroy
+    session.delete(:user_id)
+    session.delete(:cart)
+    flash[:logout] = 'Ye done walked the plank! Er, that is, ye logged out.'
```
Comment (KirbyDD) [inline app/controllers/sessions_controller.rb:19]: Love this message

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -12,8 +12,16 @@
...
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register" %>
+      <% if current_user %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Profile", "/profile" %>
+        <%= link_to "Logout", "/logout", method: :delete %>
+        <div>Logged in as <%= current_user.name %></div>
+      <% else %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
```
Comment (KirbyDD) [inline app/views/layouts/application.html.erb:23]: This seems like the same as 16 so cant it be outside the conditional?

**Review by KirbyDD — APPROVED** [review]
I think we should remove the duplicated line on that application.html but besides that looks good

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (DanielEFrampton) [reply app/views/layouts/application.html.erb:23]: I grabbed this from Melrob & Phil's user story 3 code, they have a reason for it; some user types won't be able to see the cart.

Comment (KirbyDD) [reply app/views/layouts/application.html.erb:23]: alrighty

**Review by philjdelong — APPROVED** [review]
Looking good here, explainations all make sense (pre-class metting)

**Review by mel-rob — APPROVED** [review]
Looks good. Will be sure to use role setup for 3-9.

**Comment by DanielEFrampton** [comment]
Also closes #14, closes #16, and closes #18.

**Pull request merged**